src/Compiler/CompilerUtils.cpp

```diff
@@ -239,29 +239,49 @@ struct Command {
     return *this;
   }
 
-  // Execute command.
-  void exec() {
+  // Execute command in current work directory.
+  //
+  // If the optional wdir is specified, the command will be executed
+  // in the specified work directory. Current work directory is
+  // restored after the command is executed.
+  void exec(std::string wdir = "") {
     auto argsRef = std::vector<llvm::StringRef>(_args.begin(), _args.end());
     bool verbose = false;
     if (const auto &verboseStr = getEnvVar("ONNX_MLIR_VERBOSE"))
       istringstream(verboseStr.getValue()) >> verbose;
 
+    // If a work directory is specified, save the current work directory
+    // and switch into it. Note that if wdir is empty, new_wdir will be
+    // cur_wdir.
+    SmallString<8> cur_wdir;
+    SmallString<8> new_wdir(wdir);
+    llvm::sys::fs::current_path(cur_wdir);
+    llvm::sys::fs::make_absolute(cur_wdir, new_wdir);
+    if (std::error_code ec = llvm::sys::fs::set_current_path(new_wdir)) {
+      cerr << StringRef(new_wdir).str() << ": " << ec.message() << endl;
+      exit(ec.value());
+    }
+
     // If in verbose mode, print out command before execution.
     if (verbose)
-      cout << _path << ": " << llvm::join(argsRef, " ") << "\n";
+      cout << "[" << StringRef(new_wdir).str() << "]" << _path << ": "
+           << llvm::join(argsRef, " ") << endl;
 
     std::string errMsg;
     int rc = llvm::sys::ExecuteAndWait(_path, llvm::makeArrayRef(argsRef),
         /*Env=*/None, /*Redirects=*/None,
         /*SecondsToWait=*/0, /*MemoryLimit=*/0, &errMsg);
 
     if (rc != 0) {
-      fprintf(stderr, "%s\n", llvm::join(argsRef, " ").c_str());
-      fprintf(stderr, "Error message: %s\n", errMsg.c_str());
-      fprintf(stderr, "Program path: %s\n", _path.c_str());
-      fprintf(stderr, "Command execution failed.");
+      cerr << llvm::join(argsRef, " ") << endl
```

Comment:
Suggest to use `llvm::errs()`, I see is used to emit other error messages..

Reply:
same here